crates/tauri/build.rs

```diff
@@ -236,6 +236,27 @@ fn main() {
   alias("desktop", !mobile);
   alias("mobile", mobile);
 
+  let has_default = has_feature("default");
+
+  // a workaround for default features activation based on target platform
+  // https://github.com/rust-lang/cargo/issues/1197
+  if has_default
+    && ["native-tls", "native-tls-vendored", "rustls-tls"]
+      .iter()
+      .all(|f| !has_feature(f))
+  {
+    if mobile {
+      println!("cargo:rustc-cfg=feature=\"rustls-tls\"");
+    } else {
+      println!("cargo:rustc-cfg=feature=\"native-tls\"");
+    }
+  }
+
+  // a workaround for default features activation based on profile
+  if has_default && env::var("PROFILE").as_deref() == Ok("release") {
```

Comment:
this won't work if you use `tauri build --debug`

Reply:
we can keep this here and inject `tauri/custom-protocol` in debug from CLI? this way `cargo build` and `cargo build --release` work out of the box and match `tauri dev` and `tauri build` respectively. `tauri build --debug` should be a special case that is fine to keep in CLI.

Reply:
yeah that would be fine